Sources/AsyncConnection.swift

```diff
@@ -3,7 +3,7 @@ public protocol AsyncConnection: AsyncStream {
 }
 
 extension AsyncConnection {
-    public func open(result: (Void throws -> AsyncConnection) -> Void) throws {
+    public func open(_ result: (Void throws -> AsyncConnection) -> Void) throws {
```

Comment:
I don’t like how `result` looks. Can be more clear. Maybe we should rename it to `completion` (which is, I guess, more familiar for the vast majority of Cocoa developers) and don’t substitute with `_`. So it will be like:
```swift
connection.open(completion: { … })
```
or
```swift
connection.open {
    ...
}
```